include/libopencm3/pac55xx/ccs.h

```diff
@@ -0,0 +1,148 @@
+#define PORTB                     (PAC55XX_SCC_BASE + 0x10)
+#define PORTC                     (PAC55XX_SCC_BASE + 0x14)
+#define PORTD                     (PAC55XX_SCC_BASE + 0x18)
+#define PORTE                     (PAC55XX_SCC_BASE + 0x1C)
+#define PORTF                     (PAC55XX_SCC_BASE + 0x20)
+#define PORTG                     (PAC55XX_SCC_BASE + 0x24)
+/**@}*/
+
+/* Port Pin Mux Select Registers */
+/** @defgroup pmux_sel_regs PMUXSEL register mapping.
+@ingroup system_defines
+@{*/
+#define MUXSELR(base)             MMIO32(base)
+#define PAMUXSELR                 MUXSELR(PORTA)
+#define PBMUXSELR                 MUXSELR(PORTB)
+#define PCMUXSELR                 MUXSELR(PORTC)
+#define PDMUXSELR                 MUXSELR(PORTD)
+#define PEMUXSELR                 MUXSELR(PORTE)
+#define PFMUXSELR                 MUXSELR(PORTF)
+#define PGMUXSELR                 MUXSELR(PORTG)
+#define MUXSELR_MASK              0x7
+
+/* Enum type for port function setting for type specificity. */
+typedef enum {
+  MUXSEL_GPIO                     = 0,
+  MUXSEL_AF1                      = 1,
+  MUXSEL_AF2                      = 2,
+  MUXSEL_AF3                      = 3,
+  MUXSEL_AF4                      = 4,
+  MUXSEL_AF5                      = 5,
+  MUXSEL_AF6                      = 6,
+  MUXSEL_AF7                      = 7,
+} muxsel_func_t;
+/**@}*/
+
+/* Port Pull-Up/Down Enable Registers. */
+/** @defgroup pden_regs PUEN PDEN register mapping.
+@ingroup system_defines
+@{*/
+#define PUENR(base)               MMIO32(base + 0x1C)
+#define PAPUENR                   PUENR(PORTA)
+#define PBPUENR                   PUENR(PORTB)
+#define PCPUENR                   PUENR(PORTC)
+#define PDPUENR                   PUENR(PORTD)
+#define PEPUENR                   PUENR(PORTE)
+#define PFPUENR                   PUENR(PORTF)
+#define PGPUENR                   PUENR(PORTG)
+#define PDENR(base)               MMIO32(base + 0x38)
+#define PAPDENR                   PDENR(PORTA)
+#define PBPDENR                   PDENR(PORTB)
+#define PCPDENR                   PDENR(PORTC)
+#define PDPDENR                   PDENR(PORTD)
+#define PEPDENR                   PDENR(PORTE)
+#define PFPDENR                   PDENR(PORTF)
+#define PGPDENR                   PDENR(PORTG)
+
+/* Pull Up/Down enum for type specificity. */
+typedef enum {
+  PULL_NONE                       = 0,
+  PULL_UP                         = 1,
+  PULL_DOWN                       = 2
+} pull_updown_t;
+/**@}*/
+
+/* Port Drive Strength Enable Registers. */
+/** @defgroup dsr_regs DSR register mapping.
+@ingroup system_defines
+@{*/
+#define DSR(base)                 MMIO32(base + 0x54)
+#define PADSR                     DSR(PORTA)
+#define PBDSR                     DSR(PORTB)
+#define PCDSR                     DSR(PORTC)
+#define PDDSR                     DSR(PORTD)
+#define PEDSR                     DSR(PORTE)
+#define PFDSR                     DSR(PORTF)
+#define PGDSR                     DSR(PORTG)
```

Comment:
you need better namespace prefixes here.  it pollutes too much without any clear grouping.